File: demos/pax-britannica/pax-britannica-android/assets/data/components/resources.lua
local graphics = require 'dokidoki.graphics'

background = graphics.sprite_from_image('sprites/background.png', nil, {0, 0})

loading = graphics.sprite_from_image('sprites/loading.png', nil, {113, -10})

fighter_sprites = {
  graphics.sprite_from_image('sprites/fighter_p1.png', nil, 'center'),
  graphics.sprite_from_image('sprites/fighter_p2.png', nil, 'center'),
  graphics.sprite_from_image('sprites/fighter_p3.png', nil, 'center'),
  graphics.sprite_from_image('sprites/fighter_p4.png', nil, 'center')
}

bomber_sprites = {
  graphics.sprite_from_image('sprites/bomber_p1.png', nil, 'center'),
  graphics.sprite_from_image('sprites/bomber_p2.png', nil, 'center'),
  graphics.sprite_from_image('sprites/bomber_p3.png', nil, 'center'),
  graphics.sprite_from_image('sprites/bomber_p4.png', nil, 'center')
}

frigate_sprites = {
  graphics.sprite_from_image('sprites/frigate_p1.png', nil, 'center'),
  graphics.sprite_from_image('sprites/frigate_p2.png', nil, 'center'),
  graphics.sprite_from_image('sprites/frigate_p3.png', nil, 'center'),
  graphics.sprite_from_image('sprites/frigate_p4.png', nil, 'center')
}

factory_sprites = {
  graphics.sprite_from_image('sprites/factory_p1.png', nil, 'center'),
  graphics.sprite_from_image('sprites/factory_p2.png', nil, 'center'),
  graphics.sprite_from_image('sprites/factory_p3.png', nil, 'center'),
  graphics.sprite_from_image('sprites/factory_p4.png', nil, 'center')
}

factory_light_damage_sprites = {
  graphics.sprite_from_image('sprites/factory_light_damage_1.png', nil, 'center'),
  graphics.sprite_from_image('sprites/factory_light_damage_2.png', nil, 'center'),
  graphics.sprite_from_image('sprites/factory_light_damage_3.png', nil, 'center'),
}

factory_heavy_damage_sprites = {
  graphics.sprite_from_image('sprites/factory_heavy_damage_1.png', nil, 'center'),
  graphics.sprite_from_image('sprites/factory_heavy_damage_2.png', nil, 'center'),
  graphics.sprite_from_image('sprites/factory_heavy_damage_3.png', nil, 'center'),
}

number_sprites = {
  graphics.sprite_from_image('sprites/1.png', nil, 'center'),
  graphics.sprite_from_image('sprites/2.png', nil, 'center'),
  graphics.sprite_from_image('sprites/3.png', nil, 'center'),
  graphics.sprite_from_image('sprites/4.png', nil, 'center'),
  graphics.sprite_from_image('sprites/5.png', nil, 'center')
}

a_button = graphics.sprite_from_image('sprites/a_button.png', nil, 'center')

production_layer_1 = graphics.sprite_from_image('sprites/production1.png', nil, 'center')
production_layer_2 = graphics.sprite_from_image('sprites/production2.png', nil, 'center')
production_layer_3 = graphics.sprite_from_image('sprites/production3.png', nil, 'center')

needle_sprite = graphics.sprite_from_image('sprites/needle.png', nil, 'center')
fighter_preview_sprite = graphics.sprite_from_image('sprites/fighter_outline.png', nil, 'center')
bomber_preview_sprite = graphics.sprite_from_image('sprites/bomber_outline.png', nil, 'center')
frigate_preview_sprite = graphics.sprite_from_image('sprites/frigate_outline.png', nil, 'center')
upgrade_preview_sprite = graphics.sprite_from_image('sprites/upgrade_outline.png', nil, 'center')

health_full = graphics.sprite_from_image('sprites/health_full.png', nil, 'center')
health_some = graphics.sprite_from_image('sprites/health_some.png', nil, 'center')
health_none = graphics.sprite_from_image('sprites/health_none.png', nil, 'center')

laser_sprite = graphics.sprite_from_image('sprites/laser.png', nil, 'center')
bomb_sprite = graphics.sprite_from_image('sprites/bomb.png', nil, 'center')
missile_sprite = graphics.sprite_from_image('sprites/missile.png', nil, 'center')

bubble_sprite = graphics.sprite_from_image('sprites/bubble.png', nil, 'center')
big_bubble_sprite = graphics.sprite_from_image('sprites/big_bubble.png', nil, 'center')
explosion_sprite = graphics.sprite_from_image('sprites/explosion.png', nil, 'center')
spark_sprite = graphics.sprite_from_image('sprites/spark.png', nil, 'center')

debris_sprites = {
  graphics.sprite_from_image('sprites/debris_large.png', nil, 'center'),
  graphics.sprite_from_image('sprites/debris_med.png', nil, 'center'),
  graphics.sprite_from_image('sprites/debris_small.png', nil, 'center')
}

fish_sprites = {
  graphics.sprite_from_image('sprites/fish1.png', nil, 'center'),
  graphics.sprite_from_image('sprites/fish2.png', nil, 'center'),
  graphics.sprite_from_image('sprites/fish3.png', nil, 'center'),
  graphics.sprite_from_image('sprites/fish4.png', nil, 'center'),
  graphics.sprite_from_image('sprites/fish5.png', nil, 'center'),
  graphics.sprite_from_image('sprites/fish6.png', nil, 'center'),
  graphics.sprite_from_image('sprites/fish7.png', nil, 'center'),
  graphics.sprite_from_image('sprites/fish8.png', nil, 'center')
}

title_sprite = graphics.sprite_from_image('sprites/title.png', nil, 'center')
credits_sprite = graphics.sprite_from_image('sprites/credits.png', nil, 'center')
--press_a_sprite = graphics.sprite_from_image('sprites/press_A.png', nil, 'center')

-- woot for hacks, this fixes some icky jittering when the factory moves
local gl = require 'gl'

local function smoothen(sprite)
  sprite.tex:enable()
  gl.glTexParameterf(gl.GL_TEXTURE_2D, gl.GL_TEXTURE_MIN_FILTER, gl.GL_LINEAR)
  gl.glTexParameterf(gl.GL_TEXTURE_2D, gl.GL_TEXTURE_MAG_FILTER, gl.GL_LINEAR)
  sprite.tex:disable()
end

local function tilen(sprite)
  sprite.tex:enable()
  gl.glTexParameteri(gl.GL_TEXTURE_2D, gl.GL_TEXTURE_WRAP_S, gl.GL_REPEAT)
  sprite.tex:disable()
end

for _, s in pairs(factory_sprites) do
  smoothen(s)
end
for _, s in ipairs(fish_sprites) do
  smoothen(s)
end
smoothen(production_layer_1)
smoothen(production_layer_2)
smoothen(production_layer_3)
smoothen(needle_sprite)
smoothen(fighter_preview_sprite)
smoothen(bomber_preview_sprite)
smoothen(frigate_preview_sprite)
smoothen(upgrade_preview_sprite)
smoothen(health_full)
smoothen(health_some)
smoothen(health_none)

tilen(health_full)
tilen(health_some)
tilen(health_none)

File: demos/pax-britannica/pax-britannica/src/de/swagner/paxbritannica/Resources.java
package de.swagner.paxbritannica;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.Preferences;
import com.badlogic.gdx.audio.Music;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.Texture.TextureFilter;
import com.badlogic.gdx.graphics.g2d.Sprite;
import com.badlogic.gdx.graphics.g2d.TextureAtlas;

public class Resources {

	public Sprite title = new Sprite(new Texture(Gdx.files.internal("data/spritepack/title.png")));
	public Sprite credits = new Sprite(new Texture(Gdx.files.internal("data/spritepack/credits.png")));
	
	public TextureAtlas atlas = new TextureAtlas(Gdx.files.internal("data/spritepack/packhigh.pack"));
	// public AtlasRegion region = atlas.findRegion("imagename");
	// Sprite sprite = atlas.createSprite("otherimagename");

	public Music music = Gdx.audio.newMusic(Gdx.files.internal("data/audio/music.mp3"));

	public Sprite factoryP1 = atlas.createSprite("factoryp1");
	public Sprite factoryP2 = atlas.createSprite("factoryp2");
	public Sprite factoryP3 = atlas.createSprite("factoryp3");
	public Sprite factoryP4 = atlas.createSprite("factoryp4");
	
	public Sprite factoryP1Small = atlas.createSprite("factoryp1");
	public Sprite factoryP2Small = atlas.createSprite("factoryp2");
	public Sprite factoryP3Small = atlas.createSprite("factoryp3");
	public Sprite factoryP4Small = atlas.createSprite("factoryp4");

	public Sprite fighterP1 = atlas.createSprite("fighterp1");
	public Sprite fighterP2 = atlas.createSprite("fighterp2");
	public Sprite fighterP3 = atlas.createSprite("fighterp3");
	public Sprite fighterP4 = atlas.createSprite("fighterp4");

	public Sprite bomberP1 = atlas.createSprite("bomberp1");
	public Sprite bomberP2 = atlas.createSprite("bomberp2");
	public Sprite bomberP3 = atlas.createSprite("bomberp3");
	public Sprite bomberP4 = atlas.createSprite("bomberp4");

	public Sprite frigateP1 = atlas.createSprite("frigatep1");
	public Sprite frigateP2 = atlas.createSprite("frigatep2");
	public Sprite frigateP3 = atlas.createSprite("frigatep3");
	public Sprite frigateP4 = atlas.createSprite("frigatep4");

	public Sprite debrisSmall = atlas.createSprite("debrissmall");
	public Sprite debrisMed = atlas.createSprite("debrismed");
	public Sprite debrisLarge = atlas.createSprite("debrislarge");

	public Sprite fish1 = atlas.createSprite("fish1");
	public Sprite fish2 = atlas.createSprite("fish2");
	public Sprite fish3 = atlas.createSprite("fish3");
	public Sprite fish4 = atlas.createSprite("fish4");
	public Sprite fish5 = atlas.createSprite("fish5");
	public Sprite fish6 = atlas.createSprite("fish6");
	public Sprite fish7 = atlas.createSprite("fish7");
	public Sprite fish8 = atlas.createSprite("fish8");

	public Sprite needle = atlas.createSprite("needle");

	public Sprite background = atlas.createSprite("background");

	public Sprite blackFade = atlas.createSprite("blackfade");

	public Sprite laser = atlas.createSprite("laser");
	public Sprite missile = atlas.createSprite("missile");
	public Sprite bomb = atlas.createSprite("bomb");

	public Sprite production1 = atlas.createSprite("production1");
	public Sprite production2 = atlas.createSprite("production2");
	public Sprite production3 = atlas.createSprite("production3");

	public Sprite production_tile1 = atlas.createSprite("productiontile");
	public Sprite production_tile2 = atlas.createSprite("productiontile");
	public Sprite production_tile3 = atlas.createSprite("productiontile");
	public Sprite production_tile4 = atlas.createSprite("productiontile");

	public Sprite upgradeOutline = atlas.createSprite("upgradeoutline");
	public Sprite frigateOutline = atlas.createSprite("frigateoutline");
	public Sprite bomberOutline = atlas.createSprite("bomberoutline");
	public Sprite fighterOutline = atlas.createSprite("fighteroutline");

	public Sprite healthNone = atlas.createSprite("healthnone");
	public Sprite healthSome = atlas.createSprite("healthsome");
	public Sprite healthFull = atlas.createSprite("healthfull");

	public Sprite aButton = atlas.createSprite("abutton");
	public Sprite aCpuButton = atlas.createSprite("acpubutton");
	public Sprite aPlayerButton = atlas.createSprite("aplayerbutton");
	
	public Sprite cpuButton = atlas.createSprite("cpubutton");
	public Sprite playerButton = atlas.createSprite("playerbutton");

	public Sprite cnt1 = atlas.createSprite("1");
	public Sprite cnt2 = atlas.createSprite("2");
	public Sprite cnt3 = atlas.createSprite("3");
	public Sprite cnt4 = atlas.createSprite("4");
	public Sprite cnt5 = atlas.createSprite("5");

	public Sprite spark = atlas.createSprite("spark");
	public Sprite bubble = atlas.createSprite("bubble");
	public Sprite bigbubble = atlas.createSprite("bigbubble");
	public Sprite explosion = atlas.createSprite("explosion");

	public Sprite factoryHeavyDamage1 = atlas.createSprite("factoryheavydamage1");
	public Sprite factoryHeavyDamage2 = atlas.createSprite("factoryheavydamage2");
	public Sprite factoryHeavyDamage3 = atlas.createSprite("factoryheavydamage3");
	public Sprite factoryLightDamage1 = atlas.createSprite("factorylightdamage1");
	public Sprite factoryLightDamage2 = atlas.createSprite("factorylightdamage2");
	public Sprite factoryLightDamage3 = atlas.createSprite("factorylightdamage3");

	public Sprite touchArea1 = atlas.createSprite("touchArea");
	public Sprite touchArea2 = atlas.createSprite("touchArea");
	public Sprite touchArea3 = atlas.createSprite("touchArea");
	public Sprite touchArea4 = atlas.createSprite("touchArea");
	
	public Sprite help = atlas.createSprite("help");
	public Sprite musicOnOff = atlas.createSprite("music");
	public Sprite back = atlas.createSprite("back");
	public Sprite settings = atlas.createSprite("settings");
	public Sprite checkboxOn = atlas.createSprite("checkboxon");
	public Sprite checkboxOff = atlas.createSprite("checkboxoff");

	public static Resources instance;

	public static Resources getInstance() {
		if (instance == null) {
			instance = new Resources();
		}
		return instance;
	}

	public Resources() {
		reInit();
	}

	public void reInit() {
		dispose();
		
		Preferences prefs = Gdx.app.getPreferences("paxbritannica");
		if (prefs.getInteger("antiAliasConfig", 1) == 0) {
			atlas = new TextureAtlas(Gdx.files.internal("data/spritepack/pack.pack"));
		} else {
			atlas = new TextureAtlas(Gdx.files.internal("data/spritepack/packhigh.pack"));
		}

		try {
			if (music != null) {
				music.stop();
				music.dispose();
			} 
			music = Gdx.audio.newMusic(Gdx.files.internal("data/audio/music.mp3"));
		} catch (Exception e) {
			music = Gdx.audio.newMusic(Gdx.files.internal("data/audio/music.mp3"));
		}

		factoryP1 = atlas.createSprite("factoryp1");
		factoryP2 = atlas.createSprite("factoryp2");
		factoryP3 = atlas.createSprite("factoryp3");
		factoryP4 = atlas.createSprite("factoryp4");
		
		factoryP1Small = atlas.createSprite("factoryp1");
		factoryP2Small = atlas.createSprite("factoryp2");
		factoryP3Small = atlas.createSprite("factoryp3");
		factoryP4Small = atlas.createSprite("factoryp4");

		fighterP1 = atlas.createSprite("fighterp1");
		fighterP2 = atlas.createSprite("fighterp2");
		fighterP3 = atlas.createSprite("fighterp3");
		fighterP4 = atlas.createSprite("fighterp4");

		bomberP1 = atlas.createSprite("bomberp1");
		bomberP2 = atlas.createSprite("bomberp2");
		bomberP3 = atlas.createSprite("bomberp3");
		bomberP4 = atlas.createSprite("bomberp4");

		frigateP1 = atlas.createSprite("frigatep1");
		frigateP2 = atlas.createSprite("frigatep2");
		frigateP3 = atlas.createSprite("frigatep3");
		frigateP4 = atlas.createSprite("frigatep4");

		debrisSmall = atlas.createSprite("debrissmall");
		debrisMed = atlas.createSprite("debrismed");
		debrisLarge = atlas.createSprite("debrislarge");

		fish1 = atlas.createSprite("fish1");
		fish2 = atlas.createSprite("fish2");
		fish3 = atlas.createSprite("fish3");
		fish4 = atlas.createSprite("fish4");
		fish5 = atlas.createSprite("fish5");
		fish6 = atlas.createSprite("fish6");
		fish7 = atlas.createSprite("fish7");
		fish8 = atlas.createSprite("fish8");

		needle = atlas.createSprite("needle");

		background = atlas.createSprite("background");

		blackFade = atlas.createSprite("blackfade");

		laser = atlas.createSprite("laser");
		missile = atlas.createSprite("missile");
		bomb = atlas.createSprite("bomb");

		production1 = atlas.createSprite("production1");
		production2 = atlas.createSprite("production2");
		production3 = atlas.createSprite("production3");

		production_tile1 = atlas.createSprite("productiontile");
		production_tile1.rotate90(true);
		production_tile1.rotate90(true);
		production_tile2 = atlas.createSprite("productiontile");
		production_tile2.rotate90(false);
		production_tile3 = atlas.createSprite("productiontile");
		production_tile4 = atlas.createSprite("productiontile");
		production_tile4.rotate90(true);

		upgradeOutline = atlas.createSprite("upgradeoutline");
		frigateOutline = atlas.createSprite("frigateoutline");
		bomberOutline = atlas.createSprite("bomberoutline");
		fighterOutline = atlas.createSprite("fighteroutline");

		healthNone = atlas.createSprite("healthnone");
		healthSome = atlas.createSprite("healthsome");
		healthFull = atlas.createSprite("healthfull");

		aButton = atlas.createSprite("abutton");
		aCpuButton = atlas.createSprite("acpubutton");
		aPlayerButton = atlas.createSprite("aplayerbutton");
		
		cpuButton = atlas.createSprite("cpubutton");
		playerButton = atlas.createSprite("playerbutton");

		cnt1 = atlas.createSprite("1");
		cnt2 = atlas.createSprite("2");
		cnt3 = atlas.createSprite("3");
		cnt4 = atlas.createSprite("4");
		cnt5 = atlas.createSprite("5");

		spark = atlas.createSprite("spark");
		bubble = atlas.createSprite("bubble");
		bigbubble = atlas.createSprite("bigbubble");
		explosion = atlas.createSprite("explosion");

		factoryHeavyDamage1 = atlas.createSprite("factoryheavydamage1");
		factoryHeavyDamage2 = atlas.createSprite("factoryheavydamage2");
		factoryHeavyDamage3 = atlas.createSprite("factoryheavydamage3");
		factoryLightDamage1 = atlas.createSprite("factorylightdamage1");
		factoryLightDamage2 = atlas.createSprite("factorylightdamage2");
		factoryLightDamage3 = atlas.createSprite("factorylightdamage3");

		touchArea1 = atlas.createSprite("touchArea");
		touchArea2 = atlas.createSprite("touchArea");
		touchArea3 = atlas.createSprite("touchArea");
		touchArea4 = atlas.createSprite("touchArea");
		
		help = atlas.createSprite("help");
		musicOnOff = atlas.createSprite("music");
		back = atlas.createSprite("back");
		settings = atlas.createSprite("settings");
		checkboxOn = atlas.createSprite("checkboxon");
		checkboxOff = atlas.createSprite("checkboxoff");

		title = new Sprite(new Texture(Gdx.files.internal("data/spritepack/title.png")));
		title.getTexture().setFilter(TextureFilter.Linear, TextureFilter.Linear);
		credits = new Sprite(new Texture(Gdx.files.internal("data/spritepack/credits.png")));
		credits.getTexture().setFilter(TextureFilter.Linear, TextureFilter.Linear);
	}

	public void dispose() {
		atlas.dispose();
	}

}
